lua/modicator/init.lua

```diff
@@ -12,36 +12,80 @@ end
 local options = {
   show_warnings = true, -- Show warning if any required option is missing
   highlights = {
+    overrides = {
+      bold = false,
+      bold = false
+    },
     modes = {
-      ['n']  = M.get_highlight_fg('CursorLineNr'),
-      ['i']  = M.get_highlight_fg('Question'),
-      ['v']  = M.get_highlight_fg('Type'),
-      ['V']  = M.get_highlight_fg('Type'),
-      [''] = M.get_highlight_fg('Type'),
-      ['s']  = M.get_highlight_fg('Keyword'),
-      ['S']  = M.get_highlight_fg('Keyword'),
-      ['R']  = M.get_highlight_fg('Title'),
-      ['c']  = M.get_highlight_fg('Constant'),
+      ['n'] = {
+        color = M.get_highlight_fg('CursorLineNr'),
+        bold = false,
+        italic = false,
+      },
+      ['i']  = {
+        color = M.get_highlight_fg('Question'),
+        bold = false,
+        italic = false,
+      },
+      ['v']  = {
+        color = M.get_highlight_fg('Type'),
+        bold = false,
+        italic = false,
+      },
+      ['V']  = {
+        color = M.get_highlight_fg('Type'),
+        bold = false,
+        italic = false,
+      },
+      ['�'] = {
+        color = M.get_highlight_fg('Type'),
+        bold = false,
+        italic = false,
+      },
+      ['s']  = {
+        color = M.get_highlight_fg('Keyword'),
+        bold = false,
+        italic = false,
+      },
+      ['S']  = {
+        color = M.get_highlight_fg('Keyword'),
+        bold = false,
+        italic = false,
+      },
+      ['R']  = {
+        color = M.get_highlight_fg('Title'),
+        bold = false,
+        italic = false,
+      },
+      ['c']  = {
+        color = M.get_highlight_fg('Constant'),
+        bold = false,
+        italic = false,
+      },
     },
   },
 }
 
---- Sets the foreground color value of the `CursorLineNr` highlight groups to
---- `color`.
---- @param color string
-M.set_highlight = function(color)
-  local base_highlight = api.nvim_get_hl_by_name('CursorLineNr', true)
-  local opts = vim.tbl_extend('keep', { foreground = color }, base_highlight)
-  api.nvim_set_hl(0, 'CursorLineNr', opts)
+--- Set the foreground color, bold text, and italic text
+--- of 'CursorLineNr'.
+--- @param format table
+M.set_format = function(format)
+  local ors = options.highlights.overrides
+  local args = {
+    foreground = format.color,
+    bold = has_overrides and ors.bold or format.bold,
+    italic = has_overrides and ors.italic or format.italic
```

Comment:
You seem to be using a global variable `has_overrides` to determine if the highlight should be bold and/or italic, and global variables should generally be avoided.

Perhaps we should call the option table's default bold/italic options `defaults` instead of `overrides`. Then we can use `vim.tbl_extend()` to merge the defal do something like this instead:

```lua
local args = {
  foreground = format.color,
  bold = format.bold,
  italic = format.italic
}
vim.tbl_extend('force', options.highlights.overrides, args)